lib/internal/process/stdio.js

```diff
@@ -9,8 +9,11 @@ function setupStdio() {
   var stdout;
   var stderr;
 
+  const isMainThread = process.binding('worker').threadId === 0;
```

Comment:
This `threadId === 0` check exists in a bunch of places. Could just have a getter at `process.binding('worker').isMainThread` to make it clearer what the checks are doing.

Reply:
If https://github.com/nodejs/node/pull/15759 lands I’d probably just tack it onto the internal module function wrapper :smirk_cat:

Reply:
@Qard Rebased, this is done now :)